core/src/components/input/input.tsx

```diff
@@ -432,37 +434,17 @@ export class Input implements ComponentInterface {
     this.isComposing = false;
   };
 
-  private clearTextInput = (ev?: Event) => {
+  private onClearButtonClick = (ev?: Event) => {
     if (this.clearInput && !this.readonly && !this.disabled && ev) {
       ev.preventDefault();
       ev.stopPropagation();
-
       // Attempt to focus input again after pressing clear button
       this.setFocus();
     }
-
     this.value = '';
-    this.inputCleared = true;
-
     this.ionInput.emit(ev);
-
-    /**
-     * This is needed for clearOnEdit
-     * Otherwise the value will not be cleared
-     * if user is inside the input
-     */
-    if (this.nativeInput) {
```

Comment:
This change was made 3 years ago. I am unable to replicate the original bug: https://github.com/ionic-team/ionic-framework/issues/17055 with this removed.